### ISSUE TITLE
People fields (SPUserField) set value and get value not working

### ISSUE THREAD

**Issue by kitmenke** [kitmenke/sputility#3]
Hello,

I'd like to submit 2 bug fixings and 2 new features.

1. Bug: SPUserField, it misses the "#" part (around line 1286)

```C#
this.upLevelDiv = $("#" + this.spanUserField.id + '_upLevelDiv');
this.textareaDownLevelTextBox = $("#" + this.spanUserField.id + '_downlevelTextBox');
this.linkCheckNames = $("#" +this.spanUserField.id + '_checkNames');
this.txtHiddenSpanData = $("#" +this.spanUserField.id + '_hiddenSpanData');
```

GetValue() (line 1291)

```C#
this.GetValue = function () {
     var peopleSpans = this.upLevelDiv.find('span.ms-entity-resolved');
     if (peopleSpans.length > 0) {
        return arrayToSemicolonList(peopleSpans);
     }
     return '';
};
```

2. Feature: add Description to SPField (line 377)
This expose the Description of a field as a text element.

```C#
this.Description = (function(ctls) { var c = ctls.childNodes[ctls.childNodes.length-1]; return (c.nodeType!=3 || $(c).text().replace(/\s/g,'').length == 0)? null : c; })(fieldParams.controlsCell);
```

3. Feature: registering "change" event in SPUserField (line 1334, at the bottom of SPUserField class)

```C#
var currentValue = null;
this.OnChange = function(fn) {
      var handler = getMethodName(fn) + "()";
      $(this.upLevelDiv[0], this.textareaDownLevelTextBox[0]).attr("onvaluesetfrompicker", handler);

      currentValue = this.GetValue();
      this.upLevelDiv.blur($.proxy(function() {
           if (this.GetValue() !== currentValue) {
               currentValue = this.GetValue();
               fn();
           }
      }, this));
};
```

Usage:

```C#
SPUtility.GetSPField("Nominated User").OnChange(requestUser);
function requestUser() {
    var val = SPUtility.GetSPField("Nominated User").GetValue();
    if (!val.length) return;
    alert(val);
}
```

And this method to get the list of resolved users by loginName/display Name:

```C#
this.GetUsers = function() {
    var people = [], peopleSpans = this.upLevelDiv.find('span.ms-entity-resolved');
    if (peopleSpans.length > 0) {
        $(peopleSpans).each(function () {
            var c = this.firstChild;
            people.push({ id: c.getAttribute('key'), title: c.getAttribute('displaytext') });
        });
    }
    return people;
};
```

Of course, feel free to rename the methods as you want.

_Tested on IE 8 & Chrome_

**Comment by kitmenke** [comment]
GetValue and SetValue now working with v0.11.0 in SharePoint 2007. I would like to try and come up with some sort of standard return value though. In 2007, GetValue returns a string and in 2013 an object is returned with a lot more info.

**Comment by kitmenke** [comment]
This should be resolved completely now in v0.13.0
